server/src/main/java/org/opensearch/rest/ResponseLimitBreachedException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Thrown when api response breaches threshold limit.
+ *
+ * @opensearch.internal
+ */
+public class ResponseLimitBreachedException extends OpenSearchException {
```

Comment:
We can take entity type and limit as arguments and construct the metadata fields for the message.

Ref : [cbe](https://github.com/rajiv-kv/OpenSearch/blob/71f6136db5f6fe8661a6dd5ec46ed7c6336ba63b/libs/core/src/main/java/org/opensearch/core/common/breaker/CircuitBreakingException.java#L101)

Reply:
{"error":{"root_cause":[{"type":"response_limit_breached_exception","reason":"Too many shards requested. Can not request shards beyond {1}","response_limit":1,"limit_entity":"SHARDS"}],"type":"response_limit_breached_exception","reason":"Too many shards requested. Can not request shards beyond {1}","response_limit":1,"limit_entity":"SHARDS"},"status":429}%